Evolution moteur de recherche

victorlohezic opened the pull request:
Actuellement, Elasticsearch constitue le moteur de recherche de Zeste de Savoir. Il permet d’indexer le contenu du site (provenant de la base de données MariaDB et de fichiers sur le disque) puis de rechercher dans cet index. Cette PR consiste à remplacer l'ancienne version de Elasticsearch par la dernière version de Typesense. 

On a donc mis à jour l'application searchv2 avec Typesense mais aussi les applications tutorialv2 et forum (pour générer les schémas des documents à ajouter aux collections).

### Contrôle qualité

La procédure d'installation est la suivante : 
- source zdsenv/bin/activate
- make index-all 
- make zmd-start
- make run 

Il est alors possible de faire une recherche, filtrer en fonction de : contenus publiés, messages du forum, sujets du forum. 
Si on ajoute un contenu, le modifie,  il est possible de faire make index-flagged pour l'indexer.

thibautlanier committed:
Ajout du répertoire searchv3

thibautlanier committed:
searchv3

thibautlanier committed:
Merge branch 'evolution_moteur_de_recherche' of https://github.com/victorlohezic/zds-site into evolution_moteur_de_recherche

victorlohezic committed:
Ajout script pour installer Typesense

victorlohezic committed:
Ajout du client Typesense

victorlohezic committed:
Modification du client pour ajouter quelques données et tester la recherche

victorlohezic committed:
Rendre l'application searchv3 indépendante de searchv2 sauf les tests

victorlohezic committed:
Modification pour rendre searchv3 indépendant de searchv2

victorlohezic committed:
Modification de la vue pour rechercher différents types de contenus

victorlohezic committed:
Modification des templates pour pouvoir afficher les contenus publiés obtenus en recherchant avec Typesense

Commit:
correction format des données

Commit:
ajout des fausses données pour tester le moteur de recherche

victorlohezic committed:
Ajout d'un script pour ajouter de fausses données au bon format

victorlohezic committed:
Modification pour que la recherche par post, topic, contenu publié et chapitre fonctionne

victorlohezic committed:
Ajout de la recherche dans plusieurs collections différentes

thibautlanier committed:
Début de l'indexation des données avec Typesense

thibautlanier committed:
Modification de Topic  pour l'indexation Typesense

thibautlanier committed:
Test de l'indexation sur les topics

thibautlanier committed:
Correction du format Typesense sur Topic

thibautlanier committed:
Ajout de l'indexation

victorlohezic committed:
Gestion dans la fusion des résultats de recherche du cas d'erreur

victorlohezic committed:
Modification des méthodes renvoyant des résultats pour seulement un type de collection

victorlohezic committed:
Prise en compte des filtres contenus publiés, messages et sujets du forum

thibautlanier committed:
Ajout d'une fonction qui convertit les dates

thibautlanier committed:
Ajout de l'indexation de Post

thibautlanier committed:
Correction indexation des post

victorlohezic committed:
Changement de nom des champs de la collection post

Commit:
fix du format de la date lors d'une recherche et ajout d'une commande dans le makefile pour lancer le serveur typesense

thibautlanier committed:
Ajout indexation des chapitres et publishedcontent

thibautlanier committed:
Supression des méthodes spécifiques à ES

thibautlanier committed:
Corrige la suppresion d'un contenu sur Typesense

thibautlanier committed:
Changement mapping ES en schema TS

thibautlanier committed:
Supprime les attributs relatif à Easticsearch

thibautlanier committed:
Corrige la suppression d'un document sur Typesense

thibautlanier committed:
Corrige la suppression et modification document

victorlohezic committed:
Renommage de searchv3 en searchv2

victorlohezic committed:
Modification pour permettre l'installation de Typesense

victorlohezic committed:
Modification bug affichage date pour les contenus publiés

victorlohezic committed:
Suppression de l'affichage des balises html dans la description des posts

victorlohezic committed:
Ajout d'un filtre par type de contenus

victorlohezic committed:
Ajout filtre pour les publications validées

victorlohezic committed:
Merge branch 'zestedesavoir:dev' into evolution_moteur_de_recherche

victorlohezic committed:
Mise à jour de l'intégration continue avec Typesense

victorlohezic committed:
Merge branch 'evolution_moteur_de_recherche' of github.com:victorlohezic/zds-site into evolution_moteur_de_recherche

victorlohezic committed:
Modification de l'intégration continue pour Typesense

victorlohezic committed:
Modification de l'indentation

victorlohezic committed:
Modification de l'intégration continue pour installer Typesense

victorlohezic committed:
Création d'un dossier data pour Typesense avec la ci

victorlohezic committed:
Installation de Typesense pour la ci

victorlohezic committed:
Suppression d'une erreur de syntaxe

victorlohezic committed:
Modification de l'image Typesense

victorlohezic committed:
Ajout de la variable d'environnement correspondant à la version de Typesense

victorlohezic committed:
Ajout de la version 0.24.0 de Typesense

victorlohezic committed:
Correction d'une erreur de version Typesense

victorlohezic committed:
Utilisation seulement du service Typesense

victorlohezic committed:
Changement de l'URL pour installer Typesense

victorlohezic committed:
Installation de Typesense avec le fichier binaire précompilé

victorlohezic committed:
Suppression erreur de syntaxe ligne 26

victorlohezic committed:
Changement de l'architecture du binaire

victorlohezic committed:
Lancement de Typesense en arrière plan

philippemilink commented on scripts/define_variable.sh:
```diff
@@ -14,6 +14,14 @@ if [[ $ZDS_ELASTIC_VERSION == "" ]]; then
+fi
+
+if [[ $ZDS_TYPESENSE_VERSION == "" ]]; then
+    ZDS_TYPESENSE_VERSION="xyz"
```
ça devrait être key, je pense, au lieu de version

thibautlanier committed:
Changement des noms faisant références à ES

thibautlanier committed:
Changement settings Elasticsearch vers Typesense

victorlohezic committed:
Mise à jour de la vue SuggestionContentView avec Typesense

victorlohezic committed:
Fusion de la vue SuggestionContentView pour Typesense

victorlohezic committed:
Mise à jour de la vue SuggestionContentView avec Typesense

victorlohezic committed:
Modification de la vue SimilartopicsView pour utiliser Typesense

victorlohezic committed:
Ajout de commentaires pour les vues de suggestions

thibautlanier committed:
Changement initialisation de SearchIndexManager

thibautlanier committed:
Merge branch 'evolution_moteur_de_recherche' of https://github.com/victorlohezic/zds-site into evolution_moteur_de_recherche

philippemilink commented on .github/workflows/ci.yml:
```diff
@@ -180,6 +182,13 @@ jobs:
+          tar -xzf typesense-server-${{ env.TYPESENSE_VERSION }}-linux-amd64.tar.gz
+          mkdir $(pwd)/typesense-data 
+          ./typesense-server --data-dir=$(pwd)/typesense-data --api-key=xyz --enable-cors &
+
```
Si vous déclarez un service typesense au-dessus, il y a quand même besoin de l'installer et le lancer manuellement ?

philippemilink commented on requirements.txt:
```diff
@@ -37,3 +37,6 @@ python-slugify==6.1.2
 tomli==2.0.1 ; python_version < "3.11"
+
+#Typsense 
+typesense==0.15.0
```
Pas besoin du commentaire `#Typesense`, et à placer plutôt plus haut avec les dépendances `# Explicit dependencies (references in code)` (et en profiter pour supprimer les modules ElasticSearch).

philippemilink commented on scripts/install_zds.sh:
```diff
@@ -321,6 +321,34 @@ if  ! $(_in "-elastic-local" $@) && ( $(_in "+elastic-local" $@) || $(_in "+full
+        rm typesense-server-${ZDS_TYPESENSE_VERSION}-linux-amd64.tar.gz
+        mv typesense-server-${ZDS_TYPESENSE_VERSION}-linux-amd64 typesense
+
+        export TYPESENSE_API_KEY=xyz
```
Cette variable est dans le fichier `define_variable.sh`, pas besoin de refaire un export ici.

philippemilink commented on templates/forum/topic/new.html:
```diff
@@ -37,7 +37,7 @@
         {% crispy form %}
     </div>
-
+    
```
Changement inutile, à annuler.

philippemilink commented on templates/searchv2/search.html:
```diff
@@ -33,6 +33,28 @@
+                    <li>
+
+                        {{ radio.tag }}
+
```
Ce n'est pas nécessaire de sauter partout des lignes.

philippemilink commented on templates/searchv2/search.html:
```diff
@@ -44,16 +66,16 @@
+                                {% include "searchv2/includes/post.part.html" with search_result=result.document %}
                             {% else %}
-                                {% trans "type d’objet inconnu" %} : {{ result.meta.doc_type }}
+                                {% trans "type d’objet inconnu" %} : {{ result.collection }}
```
On ne devrait jamais arriver dans ce cas. Le visiteur ne saura pas quoi faire de l'information et l'équipe technique ne sera jamais informée qu'il y a des recherches qui renvoient des types d'objets inconnus. Il faut les filtrer en amont et s'il y a des types d'objet inconnus, générer un avertissement, comme ce qui est fait [là](https://github.com/zestedesavoir/zds-site/blob/dev/zds/tutorialv2/views/statistics.py#L232).

philippemilink commented on zds/forum/models.py:
```diff
@@ -487,12 +493,12 @@ def save(self, *args, **kwargs):
 @receiver(pre_delete, sender=Topic)
-def delete_topic_in_elasticsearch(sender, instance, **kwargs):
+def delete_topic_in_search(sender, instance, **kwargs):
     """catch the pre_delete signal to ensure the deletion in ES"""
```
```suggestion
    """catch the pre_delete signal to ensure the deletion in the search engine"""
```

philippemilink commented on zds/searchv2/__init__.py:
```diff
@@ -3,17 +3,17 @@
-DEFAULT_ES_CONNECTIONS = {
+DEFAULT_SEARCH_CONNECTIONS = {
     "default": {
         "hosts": ["localhost:9200"],
```
Je crois que ce n'est pas 9200 le port par défaut de TypeSense.

philippemilink commented on zds/searchv2/__init__.py:
```diff
@@ -3,17 +3,17 @@
 
-def setup_es_connections():
+def setup_connections():
     """Create connection(s) to Elasticsearch from parameters defined in the settings.
```
```suggestion
    """Create connection(s) to the search engine from parameters defined in the settings.
```

philippemilink commented on zds/searchv2/add_content.sh:
Fichier à supprimer maintenant.

philippemilink commented on zds/searchv2/chapters.jsonl:
Fichier à supprimer maintenant.

philippemilink commented on zds/searchv2/client.py:
Fichier à supprimer maintenant.

philippemilink commented on zds/searchv2/content.jsonl:
Fichier à supprimer maintenant.

philippemilink commented on zds/searchv2/forms.py:
```diff
@@ -29,6 +29,27 @@ class SearchForm(forms.Form):
+        choices=content_type_values,
+    )
+
+    validated_content_labels = sorted(
```
Au-dessus, la variable équivalente s'appelle `content_type_values` (suffixe `values`), là vous suffixez avec `labels`. Mettez vous d'accord sur un choix unique, pour que ce soit cohérent.

philippemilink commented on zds/searchv2/management/commands/es_manager.py:
Ce fichier devrait être renommé en quelque chose comme `search_manager.py`.

philippemilink commented on zds/searchv2/management/commands/es_manager.py:
```diff
@@ -12,17 +12,17 @@ class Command(BaseCommand):
-        if not self.index_manager.connected_to_es:
-            raise Exception("Unable to connect to Elasticsearch, aborting.")
+        if not self.index_manager.connected_to_search:
+            raise Exception("Unable to connect to Typesense, aborting.")
```
```suggestion
            raise Exception("Unable to connect to the search engine, aborting.")
```
Tentons d'abstraire les choses :)

philippemilink commented on zds/searchv2/management/commands/es_manager.py:
```diff
@@ -44,13 +44,11 @@ def handle(self, *args, **options):
-
-        self.index_manager.refresh_index()
+        self.index_manager.reset_index(self.models)
+        # self.index_manager.setup_custom_analyzer()
```
En commentaire, donc à supprimer ?

philippemilink commented on zds/searchv2/models.py:
```diff
@@ -6,73 +6,69 @@
-from elasticsearch_dsl.connections import connections
-
 from django.db import transaction
+from typesense import Client
```
```suggestion
from typesense import Client as SearchEngineClient
```
pour que ce soit plus simple par la suite de comprendre de quel client on parle.

philippemilink commented on zds/searchv2/models.py:
```diff
@@ -6,73 +6,69 @@
     - ``get_indexable()`` ;
-    - ``get_mapping()`` (not mandatory, but otherwise, ES will choose the mapping by itself) ;
-    - ``get_document()`` (not mandatory, but may be useful if data differ from mapping or extra stuffs need to be done).
+    - ``get_schema()`` (not mandatory, but otherwise, ES will choose the schema by itself) ;
```
```suggestion
    - ``get_schema()`` (not mandatory, but otherwise, the search engine will choose the schema by itself) ;
```

philippemilink commented on zds/searchv2/models.py:
```diff
@@ -6,73 +6,69 @@
             Also consider the ``index='not_analyzed'`` option to improve performances.
 
-        See https://elasticsearch-dsl.readthedocs.io/en/latest/persistence.html#mappings
+        See https://elasticsearch-dsl.readthedocs.io/en/latest/persistence.html#schemas
```
Lien à supprimer ? (la note au-dessus aussi ?)

philippemilink commented on zds/searchv2/models.py:
```diff
@@ -6,73 +6,69 @@
+        es_schema = dict()
+        es_schema["name"] = self.get_document_type()
+        es_schema["fields"] = []
+        return es_schema
```
Encore trop de mentions à ElasticSearch à mon goût :)

philippemilink commented on zds/searchv2/models.py:
```diff
@@ -115,7 +112,7 @@ def get_es_document_source(self, excluded_fields=None):
+    def get_document_as_bulk_action(self, action="index"):
         """Create a document formatted for a ``_bulk`` operation. Formatting is done based on action.
 
         See https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html.
```
Encore d'actualité de faire des bulks ? (le lien n'est certainement plus pertinent !)

philippemilink commented on zds/searchv2/models.py:
```diff
@@ -215,41 +199,37 @@ def save(self, *args, **kwargs):
 
-def delete_document_in_elasticsearch(instance):
+def convert_to_unix_timestamp(date):
+    return int(time.mktime(date.timetuple()))
```
`date.timestamp()` ne fait pas le boulot ?

philippemilink commented on zds/searchv2/posts.jsonl:
Fichier à supprimer maintenant.

philippemilink commented on zds/searchv2/publishedcontent.jsonl:
Fichier à supprimer maintenant.

philippemilink commented on zds/searchv2/models.py:
```diff
@@ -418,29 +361,26 @@ def setup_custom_analyzer(self):
-        self.logger.info(f"unindex {model.get_es_document_type()}")
+        self.logger.info(f"unindex {model.get_document_type()}")
 
     def es_bulk_indexing_of_model(self, model, force_reindexing=False):
```
```suggestion
    def bulk_indexing_of_model(self, model, force_reindexing=False):
```

philippemilink commented on zds/searchv2/models.py:
```diff
@@ -6,73 +6,69 @@
+    You also need to maintain ``index_id`` and ``es_already_indexed`` for bulk indexing/updating (if any).
     """
 
     es_already_indexed = False
```
```suggestion
    already_in_search_engine = False
```

philippemilink commented on zds/searchv2/models.py:
```diff
@@ -563,65 +499,47 @@ def es_bulk_indexing_of_model(self, model, force_reindexing=False):
+            self.search.collections[doc_type].documents[doc_id].update(doc)
+            self.logger.info(f"partial_update {document.get_document_type()} with id {document.index_id}")
+        except:
+            pass
```
Qu'est-ce qui peut arriver comme exception ? On veut vraiment la passer sous silence ?

philippemilink commented on zds/searchv2/tests/tests_models.py:
```diff
@@ -26,21 +23,21 @@ def setUp(self):
+        manager = SearchIndexManager(**custom_index)
 
         # in the beginning: the void:
         self.assertTrue(manager.index not in self.manager.es.cat.indices())
```
`es` a été renommé `search` (et à de nombreuses occurrences plus bas)

philippemilink commented on zds/searchv2/models.py:
```diff
@@ -6,73 +6,69 @@
 
     es_already_indexed = False
-    es_id = ""
+    index_id = ""
```
```suggestion
    search_engine_id = ""
```
me paraît mieux, car sinon on peut ne pas savoir de quel index il s'agit.

philippemilink commented on zds/searchv2/topics.jsonl:
Fichier à supprimer maintenant.

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -15,65 +13,85 @@
 from zds.utils.templatetags.authorized_forums import get_authorized_forums
-from functools import reduce
+
+from .client import client
```
Non, il faut faire comme vous avez fait dans `zds/searchv2/models.py`.

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -15,65 +13,85 @@
         data = {"results": results}
         return HttpResponse(json_handler.dumps(data), content_type="application/json")
 
+    def _add_a_numerical_filter(self, field, values):
```
```suggestion
    def _add_numerical_filter(self, field, values):
```

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -15,65 +13,85 @@
+        """
+        Return a filter (string), this filter is used for numerical values necessary for the field
+        field : it's a string with the name of the field to filter
+        values : is a list of int with value that we want for the field
```
Tant qu'à faire, autant utiliser le format correct pour générer une belle doc : 
```
:param field: name of the field to filter
:type field: str
:param values: list of values we want for the field
:type values: List[int]
```
(un truc du genre, regarde les autres fonctions pour t'inspirer)

Aussi, on commence lentement à mettre en place du typing, donc, pourquoi ne pas avoir : 
```py
def _add_numerical_filter(self, field: str, values: List[int])

victorlohezic replied:
J'ai mis à jour les commentaires pour la documentation. Cependant par défaut Sphinx ne génère pas la documentation pour les méthodes privées. 
On peut ajouter cette ligne dans `doc/source/conf.py`  pour le faire :

```python
# -- Extension configuration -------------------------------------------------
autodoc_default_options = {'private-members': True}
```
Est-ce que je les rends publiques ou je modifie le fichier de configuration ?

philippemilink replied:
> Est-ce que je les rends publiques ou je modifie le fichier de configuration ?

Je dirais aucun des deux. Ce n'est pas grave si elles n'apparaissent pas dans la documentation celles-ci.

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -15,65 +13,85 @@
 
 class SuggestionContentView(CreateView, SingleObjectMixin):
+    """
+    Site members who are part of the staff can choose at the end of a publication to suggest another content of the site.
```
```suggestion
    Staff members can choose at the end of a publication to suggest another content of the site.
```

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -15,65 +13,85 @@
+    """
+    Site members who are part of the staff can choose at the end of a publication to suggest another content of the site.
+    When they want to add a suggestion, they write in a text field the name of the content to suggest,
+    content proposals are then made.
```
```suggestion
    content proposals are then made, using the search engine through this view.
```

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -82,55 +100,54 @@ def __init__(self, **kwargs):
 
         return HttpResponse(json_handler.dumps(data), content_type="application/json")
 
+    def _add_a_negative_numerical_filter(self, field, values):
```
```suggestion
    def _add_negative_numerical_filter(self, field, values):
```

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -82,55 +100,54 @@ def __init__(self, **kwargs):
+    def _add_a_negative_numerical_filter(self, field, values):
+        """
+        Add a filter to the current filter, this filter is used for numerical negation
+        Indeed, in 0.24.0, Typesense doesn't allow numerical negation
```
```suggestion
        Indeed, in its version 0.24.0, Typesense doesn't allow numerical negation
```

Ce n'est pas clair quel est le problème et qu'est-ce que vous voulez vraiment faire. En lisant sans trop réfléchir, j'avais compris que c'était pour filtrer suivant des nombres négatifs. En fait c'est pour exclure un nombre ? Utiliser le terme `exclude` serait sans doute alors plus parlant que `negation` (*exclude  a numerical value*).

victorlohezic replied:
On voulait bien parler de l'exclusion d'un nombre. Je vais faire la modification.

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -82,55 +100,54 @@ def __init__(self, **kwargs):
+        Add a filter to the current filter, this filter is used for numerical negation
+        Indeed, in 0.24.0, Typesense doesn't allow numerical negation
+        field : it's a string with the name of the field to filter
+        values : is a list of strings with value that we don't want for the field
```
Idem que plus haut concernant le format de ce commentaire pour générer une belle doc.

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -168,17 +188,11 @@ def get(self, request, *args, **kwargs):
     def get_queryset(self):
-        if not self.index_manager.connected_to_es:
+        if not self.index_manager.connected_to_search:
             messages.warning(self.request, _("Impossible de se connecter à Elasticsearch"))
```
```suggestion
            messages.warning(self.request, _("Impossible de se connecter au moteur de recherche."))
```

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -190,102 +204,124 @@ def get_queryset(self):
+            self.search_content_types_count = len(self.search_content_types)
 
-            models = reduce(operator.concat, models)
+            # Check if the conent must be validated
```
```suggestion
            # Check if the content must be validated
```

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -190,102 +204,124 @@ def get_queryset(self):
-            # Setting the different querysets (according to the selected models, if any)
-            part_querysets = []
-            chosen_groups = self.search_form.cleaned_data["models"]
+            # Check which collections needs to search
```
```suggestion
            # Check in which collections search is performed
```
c'est bien ce que vous voulez dire ?

victorlohezic replied:
Oui c'est bien ça, merci.

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -190,102 +204,124 @@ def get_queryset(self):
-                        models.append(settings.ZDS_APP["search"]["search_groups"][group][1])
-            else:
-                models = [v[1] for k, v in settings.ZDS_APP["search"]["search_groups"].items()]
+            # Check which type of content is needed
```
```suggestion
            # Check which content types are searched
```

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -190,102 +204,124 @@ def get_queryset(self):
 
-            for model in models:
-                part_querysets.append(getattr(self, f"get_queryset_{model}s")())
+            # Typesense Search
```
Commentaire inutile, en plus ça rajoute une mention à TypeSense dans le code.

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -190,102 +204,124 @@ def get_queryset(self):
+                result = self._choose_single_collection_method(search_collections[0])
+            else:
+                if search_collection_count == 0:
+                    search_collections = ["publishedcontent", "topic", "chapter", "post"]
```
Ce `if` pourrait plutôt être au moment où vous définissez la première valeur de `search_collections` (ligne 208). Là on est à un endroit du code où on est sur le point de lancer la requête à TypeSense, ce n'est pas le meilleur endroit pour changer des choses !

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -190,102 +204,124 @@ def get_queryset(self):
+        filter = ""
+        if self.search_content_types:
+            filter = self._add_a_filter("content_type", self.search_content_types, filter)
+            # filter += "content_type == [`TUTORIAL`, `ARTICLE`]"
```
Commentaire (et les suivants) à supprimer ?

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -297,21 +333,20 @@ def get_queryset_topics(self):
-            & Terms(forum_pk=self.authorized_forums)
-            & MultiMatch(query=self.search_query, fields=["title", "subtitle", "tags"])
-        )
+        # filter = ""
+        # filter = self._add_a_filter("forum_pk", self.authorized_forums, filter)
```
Il ne faut pas oublier de ne lister que les forums auxquels on a accès, donc à ne pas laisser en commentaires.

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -323,37 +358,54 @@ def get_queryset_posts(self):
-
         return context
 
+    def _add_a_filter(self, field, value, current_filter):
```
```suggestion
    def _add_filter(self, field, value, current_filter):
```

philippemilink commented on zds/settings/abstract_base/zds.py:
```diff
@@ -12,19 +12,23 @@
-        "hosts": ["localhost:9200"],
+        "nodes": [
+            {
+                "host": "localhost",  # For Typesense Cloud use xxx.a1.typesense.net
+                "port": "8108",  # For Typesense Cloud use 443
+                "protocol": "http",  # For Typesense Cloud use https
```
On s'en fout de Typesense Cloud :)

philippemilink commented on zds/settings/abstract_base/zds.py:
```diff
@@ -12,19 +12,23 @@
-    "shards": 3,
-    "replicas": 0,
-}
+SEARCH_INDEX = {}
```
Cet objet semble inutile maintenant, non ?

philippemilink commented on zds/settings/prod.py:
```diff
@@ -143,7 +143,7 @@ def _get_version():
 
 
-ES_SEARCH_INDEX["shards"] = config["elasticsearch"].get("shards", 3)
+SEARCH_INDEX["shards"] = config["elasticsearch"].get("shards", 3)
```
Inutile.

philippemilink reviewed (changes requested):
C'est un gros morceau ! :)

Ça amène donc forcément beaucoup de commentaires.

Je trouve qu'il y a encore beaucoup de mentions à ElasticSearch dans le code (j'en ai pointées quelques unes, il y en a d'autres). Faites
```sh
git grep -Irni elastic
git grep -Irni shards
git grep -Irni replica
git grep -Irni es_ # un peu plus compliqué à faire le tri, mais on en voit
```
Et vous aurez un bon aperçu de ce qu'il reste à supprimer / adapter.

De façon générale, dans le code, évitez les noms de variables qui peuvent être utilisées dans un autre module que celui de la recherche et qui peut prêter à confusion. Par exemple : 
- `index_manager` -> `search_engine_manager`
- `index_id` -> `search_engine_id`
- ... et probablement d'autres dont je ne me souviens plus.

Vous faites des doc-strings pour les nouvelles fonctions et c'est bien. Mais tant qu'à faire, utilisez le format pour que Sphinx génère une belle documentation, comme on le fait déjà dans certaines fonctions, et sinon c'est (un peu) documenté [là](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html).

Les docs-strings, mais vous pouvez aussi utiliser les [type hints](https://docs.python.org/3/library/typing.html) en Python maintenant.

Concernant les classes liées à la recherche, je ne comprends pas bien pourquoi certaines ont Django dans leur nom, par exemple `AbstractSearchDjangoIndexable` (je sais, c'était le cas avec ES, mais pendant qu'on change la recherche, autant tout faire). Ça donne l'impression qu'elles utilisent une fonctionnalité de Django (voire qu'elles sont fournies par Django), alors qu'il ne me semble pas ? Ce serait bien de supprimer Django du nom de ces classes.

N'oubliez pas d'adapter aussi la documentation (on en avait parlé à notre dernière réunion).

La CI ne passe pas, il faudrait corriger ça.

Et un petit rebase sur la branche `dev` ne ferait pas de mal, pour avoir les commits qui ont été pushés dans le dépôt entre temps.

EDIT : je viens de tester en local. Ça semble fonctionner. Je remarque par contre quelques problèmes : 
- il manque les tags dans les résultats de la recherche
- dans les résultats de la recherche, lorsqu'un contenu n'a pas de miniature, on a un carré gris : on pourrait plutôt avoir la miniature par défaut des contenus, comme sur la page d'accueil (c'est déjà le comportement actuellement avec ES, mais puisqu'on est en train de revoir la recherche, autant corriger aussi ça en passant)
- si TypeSense ne peut pas être joint (coupez TypeSense, faites une recherche), on se prend une erreur pas jolie (qui donnerait une 500 en production). Il faut capturer cette erreur, mettre un joli message d'erreur et logger un avertissement pour que ce soit remonté dans notre système de supervision.

victorlohezic committed:
Suppression de changements inutiles

victorlohezic committed:
Suppression des références SEARCH_INDEX

victorlohezic committed:
Suppression de fichiers inutiles

victorlohezic committed:
Changement du nom de certaines fonctions et de certains commentaires

victorlohezic committed:
Modification des commentaires

victorlohezic committed:
Modification de la place d'un if pour un code plus stable

philippemilink commented on zds/settings/prod.py:
```diff
@@ -143,9 +143,6 @@ def _get_version():
 # ZESTE DE SAVOIR SETTINGS
 
 
-ES_SEARCH_INDEX["shards"] = config["elasticsearch"].get("shards", 3)
```
Ici il faudrait mettre : 
```py
SEARCH_CONNECTIONS["default"]["api_key"] = config["typesense"].get("api_key", "xyz")
```
Pour que lors du déploiement de Zeste de Savoir, la clé d'API soit récupérée d'un fichier de configuration externe au dépôt Git ([celui-là](https://github.com/zestedesavoir/ansible-zestedesavoir/blob/main/roles/app/templates/config.toml.j2)).

(et d'autres clés d'API si on utilise plusieurs clés avec des permissions différentes)

philippemilink commented on scripts/define_variable.sh:
```diff
@@ -10,8 +10,12 @@ if [[ $ZDS_NVM_VERSION == "" ]]; then
-if [[ $ZDS_ELASTIC_VERSION == "" ]]; then
-    ZDS_ELASTIC_VERSION="5.5.2"
+if [[ $ZDS_TYPESENSE_VERSION == "" ]]; then
+    ZDS_TYPESENSE_VERSION="0.24.0"
```
On vient de  passer à 0.24.1 :)

philippemilink commented on scripts/install_zds.sh:
```diff
@@ -321,6 +321,33 @@ if  ! $(_in "-elastic-local" $@) && ( $(_in "+elastic-local" $@) || $(_in "+full
+        mv typesense-server-${ZDS_TYPESENSE_VERSION}-linux-amd64 typesense
+
+        mkdir -p $typesense_path/typesense-data
+        ./$typesense_path/typesense-server --data-dir=$(pwd)/typesense-data --api-key=$ZDS_TYPESENSE_API_KEY --enable-cors
```
Puisqu'on n'interrogera pas TypeSense depuis des requêtes JavaScript, le paramètre `--enable-cors` ne semble pas nécessaire (cf la [doc](https://typesense.org/docs/0.24.1/api/server-configuration.html#using-command-line-arguments)).

victorlohezic committed:
Ajout du filtre pour afficher les sujets et messages des forums autorisés

victorlohezic committed:
Tentative d'ajout d'un score pour les contenus publiés

thibautlanier committed:
Correction indexation publishedcontent avec score

thibautlanier committed:
Efface des mentions à  Elasticsearch

thibautlanier committed:
Suprrime les référence à Django dans la recherche

victorlohezic committed:
Ajout des fonctions pour calculer les scores

victorlohezic committed:
Merge branch 'evolution_moteur_de_recherche' of github.com:victorlohezic/zds-site into evolution_moteur_de_recherche

thibautlanier committed:
Changement de noms d'attributs et fonctions Typesense

thibautlanier committed:
Merge branch 'evolution_moteur_de_recherche' of https://github.com/victorlohezic/zds-site into evolution_moteur_de_recherche

thibautlanier committed:
Ajout de setup_search

victorlohezic committed:
Prise compte du score précalculé dans la recherche

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -228,28 +228,27 @@ def get_queryset(self):
                     "query_by": "title,description,categories,subcategories, tags, text",
-                    "query_by_weights": "10,2,2,2,2,2",
-                    # "sort_by": "score:asc"
+                    "query_by_weights": f"{settings.ZDS_APP['search']['boosts']['publishedcontent']['title']},{settings.ZDS_APP['search']['boosts']['publishedcontent']['description']},{settings.ZDS_APP['search']['boosts']['publishedcontent']['categories']},{settings.ZDS_APP['search']['boosts']['publishedcontent']['subcategories']},{settings.ZDS_APP['search']['boosts']['publishedcontent']['tags']} ...
```
C'est difficilement lisible, et je pense que ça ne plaira pas à `blake` parce que la ligne est bien trop longue (tente de lancer `make lint-black`).

Tu peux faire quelque chose du genre : 
```py
content_boosts = settings.ZDS_APP['search']['boosts']['publishedcontent']
# ...
"query_by_weights": "{},{},{},{}".format(
    content_boosts['title'],
    content_boosts['description'],
    # ...
)
```
(idem plus bas)

victorlohezic replied:
C'est vrai que c'était pas top pour la lisibilité, j'ai apporté la modification (pour info make lint-back ne posait pas de problème).

victorlohezic committed:
Amélioration de la lisibilité dans l'utilisation des poids

victorlohezic committed:
Suppression des références à Elasticsearch

victorlohezic committed:
Utilisation du client Typesense standardisée

victorlohezic committed:
Mise en forme des commentaires pour la doc

victorlohezic committed:
Suppression problème d'indentation pour la documentation

philippemilink commented:
Sentry rapporte que l'erreur suivante a lieu lorsqu'on tente de chercher `*` : 
```
KeyError: 'text_match'
  File "django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
  File "zds/searchv2/views.py", line 188, in get
    return super().get(request, *args, **kwargs)
  File "django/views/generic/list.py", line 142, in get
    self.object_list = self.get_queryset()
  File "zds/searchv2/views.py", line 256, in get_queryset
    result = self.get_queryset_multisearch(search_requests, search_collections)
  File "zds/searchv2/views.py", line 273, in get_queryset_multisearch
    all_collection_result.sort(key=lambda result: result["text_match"], reverse=True)
  File "zds/searchv2/views.py", line 273, in <lambda>
    all_collection_result.sort(key=lambda result: result["text_match"], reverse=True)
```

Je viens de le reproduire sur le serveur bêta. Ça a pu être corrigé avec les derniers commits, mais je préfère le signaler.

victorlohezic committed:
Résolution de bugs

philippemilink commented on zds/searchv2/views.py:
```diff
@@ -328,17 +328,18 @@ def get_queryset_multisearch(self, search_requests, collection_names):
-                    reverse=True,
-                )
-                print(all_collection_result[i * 10 : (i + 1) * 10])
+        if self.search_query != "*":
```
Je suis sceptique. J'allais dire que le bug se produit sans doute aussi avec d'autres caractères spéciaux, mais je n'ai pas réussi à en trouver. D'ailleurs, ça fonctionne si on peut un espace avant ou après l'astérisque. L'astérisque a une signification particulière pour TypeSense ? 

Une meilleure façon (à mon sens) de gérer ça serait plutôt de vérifier que `text_match` est bien présent. Qu'est-ce que ça veut dire s'il n'est pas présent ?

victorlohezic replied:
J'ai fait cette correction car d'après la documentation Typesense : 
> Use * as the search string to return all documents. 

Pour moi, c'était le seul cas où il y a un sens à ne pas avoir `text_match`. Je n'ai pas trouvé non plus d'autres caractères spéciaux que l'on peut mettre seul.
C'est vrai que s'il y a d'autres cas dont nous n'avons pas connaissance, vérifier que text_match est bien présent permet au moteur de recherche d'être plus robuste.

philippemilink replied:
Ok. Dans ce cas, il faudrait faire les choses suivantes : 
- ajouter un commentaire qui explique que `*` permet de retourner tous les documents avec TypeSense (et éventuellement mettre un lien vers la doc qui le dit)
- qu'est-ce qu'on fait si `*` est tout de même recherché ?  j'ai envie de dire de n'afficher aucun résultat (voire un message plus explicite comme `Recherche invalide.`)
- ajouter un test qui recherche `*`
- ce serait quand même bien de ne garder que les résultats qui ont un `text_match`, pour s'assurer que ce point ne nous pétera jamais à la figure. C'est faisable en conditionnant les deux lignes du `for` juste au-dessus par une condition `if 'text_match' in entry:`.

philippemilink replied:
Ah, je vois que vous vous servez dans les tests du fait que chercher `*` renvoie tout ce qui est indexé. Autant pour les tests, ça ne me gêne pas, autant en production la possibilité de demander à TypeSense de renvoyer tout ce qu'il a m'inquiète car c'est inutile et ça va utiliser beaucoup de ressource (bande passante, traitement des résultats, etc). À l'extrême on pourrait presque considérer comme une possible attaque par déni de service. Est-ce qu'il n'y a pas un autre moyen dans les tests de demander à TypeSense qu'il retourne tout ce qu'il a indexé ?

victorlohezic committed:
Suppression des lignes pour debug

victorlohezic committed:
Utilisation du bon poids pour les chapitres

thibautlanier committed:
Merge branch 'evolution_moteur_de_recherche' of https://github.com/victorlohezic/zds-site into evolution_moteur_de_recherche

thibautlanier committed:
Correction de la méthode delete_by_query

thibautlanier committed:
Ajout des tests_models pour Typesense

victorlohezic committed:
Correction du lien des tags

victorlohezic committed:
Merge branch 'evolution_moteur_de_recherche' of github.com:victorlohezic/zds-site into evolution_moteur_de_recherche

victorlohezic committed:
Affichage des sous-titres en entier

thibautlanier committed:
Supprimme des références à Typesense

thibautlanier committed:
Modification documentation de fonctions

thibautlanier committed:
Merge branch 'evolution_moteur_de_recherche' of https://github.com/victorlohezic/zds-site into evolution_moteur_de_recherche

thibautlanier committed:
Suppresion de setup_custom_analyzer

thibautlanier committed:
Changement de bulk

thibautlanier committed:
Changement de get_document_schema

thibautlanier committed:
Ajout de datetime.timestamp pour la conversion

victorlohezic committed:
Réécriture des trois premiers tests pour la vue

victorlohezic committed:
Merge

victorlohezic committed:
Réécriture des trois premiers tests de la vue

victorlohezic committed:
Réécriture d'un test supplémentaire de la vue

victorlohezic committed:
Rédaction de la présentation de Typesense et de la phase de recherche

victorlohezic committed:
Correction typographie

victorlohezic committed:
Multiplication de text_match avec le score précalculé

victorlohezic committed:
Ajustement des poids pour la recherche

victorlohezic committed:
Traitement de la recherche avec *

philippemilink commented on templates/searchv2/search.html:
```diff
@@ -51,7 +51,11 @@
-            {% if object_list %}
+            {% if object_list == '*' %}
+                <p>
+                    Recherche invalide.
```
```suggestion
                    {% trans "Recherche invalide." %}
```
Et il faut aussi traduire la chaîne dans le test correspondant.

victorlohezic committed:
Tentative d'une nouvelle configuration et explications

victorlohezic committed:
Début implementation highlight

thibautlanier committed:
Modification de clear_index

thibautlanier committed:
Suppression des mentions à SEARCH_INDEX

thibautlanier committed:
Supprime des références à elasticsearch

thibautlanier committed:
Merge branch 'evolution_moteur_de_recherche' of https://github.com/victorlohezic/zds-site into evolution_moteur_de_recherche

thibautlanier committed:
Evite la suppresion et ajout de doublons de collections

thibautlanier committed:
Modification de tests_utils pour typesense

thibautlanier committed:
Début de documentation pour l'installation de Typesense

thibautlanier committed:
Supprime références à Elasticsearch

thibautlanier committed:
Ajout de la phase d'indexation dans la documentation

thibautlanier committed:
Correction  vérifier la connexion à Typesense

thibautlanier committed:
Evite l'indexation de code html

victorlohezic committed:
Modification des paramètres de recherche

victorlohezic committed:
Traitement du cas où text de clean_html est nul

thibautlanier committed:
Documentation installation Typesense

thibautlanier committed:
Ajout documentation de fonction

victorlohezic committed:
Modification de la fonction qui nettoie l'HTML pour l'indexation

victorlohezic committed:
Merge branch 'evolution_moteur_de_recherche' of github.com:victorlohezic/zds-site into evolution_moteur_de_recherche

victorlohezic committed:
Réécriture des tests de boosts et d'impacts sur les messages de forum

victorlohezic committed:
Réécriture du tests de le recherche d'une même requête en majuscule et minuscule

victorlohezic committed:
Réécriture du test sur les chapitres

victorlohezic committed:
Réécriture du test de la vue concernant les catégories et sous-catégories

Commit:
add highlight for forum topics and messages and change highlight style to match previous one

Commit:
Resolution d'un bug lié au formattage des dates

Commit:
Changement code format date

philippemilink committed:
Merge branch 'dev' into evolution_moteur_de_recherche

philippemilink committed:
Ajoute les migrations manquantes

(à renommer et squasher avant de merger la PR)

philippemilink committed:
Utilisation l'action Typesense pour le lancer dans la CI

philippemilink committed:
Merge branch 'dev' into evolution_moteur_de_recherche

philippemilink committed:
Met à jour Typesense vers la version 0.24.1

philippemilink committed:
Annule des modifications inutiles

philippemilink committed:
Amélioration de la documentation de la recherche

philippemilink committed:
Merge branch 'dev' into evolution_moteur_de_recherche

philippemilink committed:
Renomme les migrations et applique black

philippemilink committed:
Adapte la documentation pour l'installation de Typesense

philippemilink committed:
Utilise la version 0.15.1 du module Python typesense

philippemilink committed:
Utilisation d'espaces insécables dans la documentation de la recherche

Juste avant les deux-points et les points-virgules.

philippemilink committed:
Utilise search_engine_id plutôt que directement les valeurs dans les tests

search_engine_id est égal à la pk de l'entité, mais on ne peut pas prévoir
sa valeur, en particulier avec MySQL.

philippemilink committed:
Retourne aussi les chapitres de tutoriels dans la recherche

philippemilink committed:
Prend en compte les chapitres dans les tests de la recherche

philippemilink committed:
Merge branch 'dev' into evolution_moteur_de_recherche

philippemilink committed:
Surligne les résultats de la recherche aussi pour les chapitres

philippemilink committed:
Corrige l'affichage de la date des résultats de la recherche

philippemilink committed:
Améliore la gestion du cas où * est recherché

philippemilink committed:
Renomme convert_to_unix_timestamp en date_to_timestamp_int

philippemilink committed:
Renomme les méthodes _compute_score() en _compute_search_score()

philippemilink committed:
Relecture de zds/forum/models.py

philippemilink committed:
Relecture de zds/searchv2/forms.py

philippemilink committed:
Ne permet qu'une unique configuration de connexion à Typesense

philippemilink committed:
Relecture de zds/searchv2/models.py

philippemilink committed:
Relecture de zds/searchv2/tests/*

philippemilink committed:
Relecture de zds.searchv2.views.SimilarTopicsView

- Déplacement du nombre de topics similaires à afficher dans
  settings.ZDS_APP
- Ajout des poids sur les différents champs des topics

philippemilink committed:
Teste que les topics similaires trouvés ne sont pas dans des forums privés

philippemilink committed:
Ajoute une classe SearchFilter pour abstraire la gestion des filtres Typesense

philippemilink committed:
Merge branch 'dev' into evolution_moteur_de_recherche

philippemilink committed:
Merge branch 'dev' into evolution_moteur_de_recherche

philippemilink committed:
Renomme une migration

philippemilink committed:
Mise à jour de Typesense

philippemilink committed:
Suppression d'un log inutile

philippemilink committed:
Protège la recherche de sujets similaire de '*'

philippemilink committed:
Relecture de zds.searchv2.views.SuggestionContentView et ajout de tests

philippemilink committed:
Rend les tests de la recherche moins verbeux

philippemilink committed:
Déplace les paramètres de recherche des collections dans les modèles

philippemilink committed:
Relecture de SearchView

philippemilink committed:
Enlève un champ inutilisé de SearchForm

philippemilink committed:
Corrige le test de SuggestionView

philippemilink committed:
Simplifie le surlignage des termes recherchés

philippemilink committed:
Relecture de zds/tutorialv2/models/database.py

philippemilink committed:
Corrige un TODO dans la documentation

philippemilink committed:
Relecture de la documentation et le script d'installation de Typesense

philippemilink committed:
Corrige un commentaire dans la CI mentionnant ElasticSearch

philippemilink committed:
Teste si les collections existent avant de chercher dedans

philippemilink committed:
Enlève les filtres avancés de la recherche

Ils n'étaient pas présents dans la version ElasticSearch et il faudrait les implémenter d'une façon différente.

philippemilink committed:
Simplifie les filtres numériques de Typesense

Possible apparemment seulement depuis la version 0.25.2

philippemilink committed:
Ne garde qu'un champ pour marquer le besoin de réindexer

philippemilink committed:
Corrige l'erreur de génération de la documentation

philippemilink committed:
Déplace SearchIndexManager et l'utilise comme unique point d'entrée pour Typesense

philippemilink committed:
Supprime quelques occurrences restantes de search_engine_already_indexed

philippemilink committed:
Simplifie la commande search_engine_manager

philippemilink committed:
Supprime AbstractSearchIndexable.get_document_for_indexing()

philippemilink committed:
Renomme get_document_schema() et get_document_type()

philippemilink committed:
Supprime la fonction delete_document_in_search_engine()

Et utilise directement directement delete_document()

philippemilink committed:
N'indexe pas les messages de forum masqués

philippemilink committed:
Ajoute les tags aux topics dans la recherche et utilise leurs slugs

philippemilink committed:
Utilise SearchIndexManager comme un singleton

philippemilink committed:
Déplace la sélection des collections où chercher dans le formulaire

philippemilink committed:
Met à jour Typesense

philippemilink committed:
Corrige un test

philippemilink committed:
Renomme 'score' en 'weight'

Sinon on pourrait croire que score est le score de pertinence des
résultats d'une recherche.

philippemilink committed:
Supprime la fonction inutilisée sub_tag()

philippemilink committed:
Supprime l'attribut PublishableContent.must_reindex

Inutilisé depuis bien longtemps, même avant le passage à Typesense.

philippemilink committed:
Marque à indexer les contenus lors de changements dans les attributs indexés

Par exemple, les tags d'un topic sont indexés, si un tag est supprimé ou
ajouté à un topic, il faut reindexer ce topic pour prendre en compte le
changement dans les tags.

philippemilink committed:
Affiche la miniature par défaut dans les résultats de la recherche

philippemilink committed:
Merge branch 'dev' into evolution_moteur_de_recherche

philippemilink committed:
Merge branch 'dev' into evolution_moteur_de_recherche

philippemilink committed:
Augmente le timeout pour parler à Typesense

philippemilink committed:
Améliore l'aide de la commande search_index_manager

philippemilink committed:
Retravaille la verbosité de la commande search_engine_manager

philippemilink committed:
Corrige la mise à jour des enregistrements dans Typesense

philippemilink committed:
Utilise une exponentiation pour adapter la taille des batchs à indexer

philippemilink committed:
Renomme le module "searchv2" en "search"

philippemilink committed:
Supprime les topics et posts de Typesense lors du déplacement vers un forum privé

philippemilink committed:
Supprime la méthode inutilisée SearchIndexManager.update_single_document()

philippemilink committed:
Corrige une mauvaise utilisation de SearchFilter.add_exact_filter()

philippemilink committed:
Gère le cas de la suppression dans Typesense alors que les collections ne sont pas créées

philippemilink committed:
Augmente encore le timeout pour parler à Typesense

philippemilink committed:
Corrige une typo qui empêchait de réduire la taille du batch

philippemilink committed:
Ne change pas la taille de batch après le dernier batch indexé

Il est fort probable que le dernier batch soit plus petit que la taille
de batch demandé, ce qui peut donner l'impression que c'est plus lent.

philippemilink committed:
Merge branch 'dev' into evolution_moteur_de_recherche

artragis commented on zds/search/utils.py:
```diff
@@ -239,15 +239,16 @@ def import_documents(objects, doc_type):
                                 objects_per_batch *= 2
                             else:
-                                objects_per_batch >> 1  # use >> 1 instead of / 2 to keep an int
+                                objects_per_batch >>= 1  # use >> 1 instead of / 2 to keep an int
```
De ce fait pour que ça soit plus clair autant utiliser l'opérateur "division entière" : `objects_per_batch //= 2`

philippemilink committed:
Simplifie la division entière par 2

philippemilink committed:
Affiche la durée d'indexation

philippemilink committed:
N'applique pas de timeout pour Typesense dans les commandes

philippemilink committed:
Indexe uniquement les champs recherchés et simplifie le code

philippemilink committed:
Indexe les symboles '+' et '#'

Pour pouvoir chercher c++ et c#

philippemilink committed:
Merge branch 'dev' into evolution_moteur_de_recherche

philippemilink committed:
Loggue les erreurs renvoyées par Typesense lors d'une recherche

philippemilink committed:
Met à jour Typesense vers la version 0.27

philippemilink committed:
Ne renvoie que 250 résultats de chaque collection lors d'une recherche

philippemilink committed:
Remplace "topic" par "sujet"

philippemilink committed:
Corrige l'affichage dans le HTML du boost des résultats de recherche

Follow-up of ca06cc927697bbb7457a31e6421f3724eeb7ce3d

philippemilink committed:
Merge branch 'dev' into evolution_moteur_de_recherche

philippemilink committed:
Merge branch 'dev' into evolution_moteur_de_recherche

philippemilink committed:
Merge branch 'dev' into evolution_moteur_de_recherche
